bluesky/examples.py

```diff
@@ -516,16 +591,93 @@ def stop(self):
         pass
 
 
+class GeneralReaderWithFileStore(Reader):
+    """
+
+    Parameters
+    ----------
+    name : string
+    read_fields : dict
+        Mapping field names to functions that return simulated data. The
+        function will be passed no arguments.
+    conf_fields : dict, optional
+        Like `read_fields`, but providing slow-changing configuration data.
+        If `None`, the configuration will simply be an empty dict.
+    monitor_intervals : list, optional
+        iterable of numbers, specifying the spacing in time of updates from the
+        device (this applies only if the ``subscribe`` method is used)
+    loop : asyncio.EventLoop, optional
+        used for ``subscribe`` updates; uses ``asyncio.get_event_loop()`` if
+        unspecified
+    fs : FileStore
+        FileStore object that supports inserting resource and datum documents
+
+    """
+
+    def __init__(self, *args, fs, save_path=None, save_func=np.save,
+                 save_spec='RWFS_NPY', save_ext='.npy',
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fs = fs
+        self.save_func = save_func
+        self.save_ext = save_ext
+        self._resource_id = None
+        if save_path is None:
+            self.save_path = mkdtemp()
+        else:
+            self.save_path = save_path
+        self.filestore_spec = save_spec  # spec name stored in resource doc
+
+        self._file_stem = None
+        self._path_stem = None
+        self._result = None
+
+    def stage(self):
+        self._file_stem = short_uid()
+        self._path_stem = os.path.join(self.save_path, self._file_stem)
+        self._resource_id = self.fs.insert_resource(self.filestore_spec,
+                                                    self._path_stem, {})
+
+    def trigger(self):
+        # save file stash file name
+        self._result = {}
```

Comment:
It's better to use `self.result.clear()` so that anything (say, in a subclass) that refers to `self._result` does not lose its reference.